### DISCUSSION TITLE
Saving as jpg compressed tiff using MagickWand

### DISCUSSION THREAD

**Discussion by kilgroem** [ImageMagick/ImageMagick#3154] [Help] [answered]
I am trying to use a c program and MagickWand to open a jpg, edit it, and write it as a tiff with jpg compression. Calling `MagickSetImageCompression(image, JPEGCompression)` doesn't produce a jpeg compressed tiff if setting `MagickSetImageFormat(image, "tif")` or using a .tif output file extension. Setting format or extension as jpg does produce a jpg whether or not I set the compression, so I'm not sure if setting compression is doing anything. Setting the compression to `Group4Compression` also does not seem to have an effect.

Converting the image on the command line works, `convert in.jpg -compress jpeg out.tiff`, so I can export then convert separately.

Any guesses what I am doing wrong here?

**Comment by kilgroem** [comment, marked as answer]
Sorry, I guess I needed `MagickSetCompression` not `MagickSetImageCompression`, although that doesn't really clarify things for me. Or why `MagickSetImageFormat` and `MagickSetFormat` both seem to have the desired effect.

**Comment by urban-warrior** [comment]
MagickSet*() methods sets an attribute in a global structure that applies to all images, whereas MagickSetImage*() sets attributes specific to a particular image.
